### ISSUE TITLE
Avoid splitting up API docs

### ISSUE THREAD

**Issue by astrofrog** [astropy/specutils#678]
There are currently three separate:

```
.. automodapi:: specutils
```

sections, each having to exclude/include classes, which gets messy. In fact SpectralAxis was included in all three sections with different options which caused intermittent docs failures.

As a user, I found it very confusing to have the API docs for a given sub-package split up in this way, and the API docs for the top-level package should really be on the front page of the docs.

**Comment by eteq** [comment]
I agree.  I think the logical mapping of sections in the Reference/API made sense at one point, but it's now gotten too find-grained so there's a lot of overlap.

So I can see two solutions to this:

1. Follow an Astropy-like route of creating sections for sub-packages.  That would involve re-organizing the docs to be in a "base" (i.e. `from specutils import X` objects), "analysis" (`from specutils.analysis import X` ), "manipulation" (`from specutils.manipulation import X` ), and "fitting" (`from specutils.fitting import X` ), and move the narrative docs around to fall in the relevant sections.
2. Change to a single global "Reference/API" section at the bottom of the docs.  Then that page would just get `automodapi` sections for the 4 cases I listed above.

I think 2 may be better because then there's just one place to look for all of this and draws a clearer distinction between "narrative" and "reference.  But I could live with either.
